src/Compiler/Facilities/DiagnosticsLogger.fs

```diff
@@ -875,17 +875,17 @@ type StackGuard(maxDepth: int, name: string) =
             [<CallerFilePath; Optional; DefaultParameterValue("")>] path: string,
             [<CallerLineNumber; Optional; DefaultParameterValue(0)>] line: int
         ) =
-        use _ =
-            Activity.start
```

Comment:
Can we do the argument creation conditionally (only if there is a listener) as well please?
Do avoid stringification of the arguments as well as the array creation.

Reply:
I am fine if the work is done in case of telemetry being collected - in that case, I don't mind either version.
I would just want to reduce the allocation overhead (this happens every single nesting, not just when new thread would spin off) if there is no active listener to the activity source.